sycl/source/detail/queue_impl.hpp

```diff
@@ -388,6 +388,10 @@ class queue_impl {
   template <typename Param>
   typename Param::return_type get_backend_info() const;
 
+  /// Begin to execute previously issued commands on this queue
+  /// if they have not been executed yet. Overrides normal batching behaviour.
+  void flush() { getPlugin()->call<PiApiKind::piQueueFlush>(MQueues[0]); }
```

Comment:
I think it is possible for MQueues to be empty. At least internally we most likely use  host queues which don't have any PI handles.

So, it will be nice to have some tests for this feature.

Reply:
Ok, I'll add a check to verify that MQueues is not empty. I'm also working on adding some tests on this PR.